app/main/views/suppliers.py

```diff
@@ -185,6 +195,12 @@ def create_supplier():
         ['contactInformation']
     )
 
+    # duns number does not have a unique constraint due to nulls
```

Comment:
First DB I've used that allows this. Nice (I think)

- Added the unqiue constraint and update the POST method to not load up to check.

Kept the other stuff to allow fail fast on the create supplier flow. We can chat to Cath tomorrow to see if she wants that or whether failing on create would be OK.